aqt/archives.py

```diff
@@ -38,6 +49,358 @@ def __init__(self, version, target, arch, os_name):
+            return ver_ext[0]
+
+        versions_extensions = ListCommand.get_versions_extensions(
+            self.fetch_http(self.archive_id.to_url()), self.archive_id.category
+        )
+        versions = sorted(
+            filter(None, map(get_version, filter(filter_by, versions_extensions)))
+        )
+        iterables = itertools.groupby(versions, lambda version: version.minor)
+        return ListCommand.Versions(iterables)
+
+    def fetch_latest_version(self) -> Optional[Version]:
+        return self.fetch_versions().latest()
+
+    def fetch_tools(self) -> Tools:
+        html_doc = self.fetch_http(self.archive_id.to_url())
+        return ListCommand.Tools(list(ListCommand.iterate_folders(html_doc, "tools")))
+
+    def fetch_tool_modules(self, tool_name: str) -> ListOfStr:
+        rest_of_url = self.archive_id.to_url() + tool_name + "/Updates.xml"
+        xml = self.fetch_http(rest_of_url)  # raises RequestException
+        modules = xml_to_modules(
+            xml,
+            predicate=ListCommand._has_nonempty_downloads,
+            keys_to_keep=(),  # Just want names
+        )
+        return ListCommand.ListOfStr(strings=list(modules.keys()))
+
+    def _to_version(self, qt_ver: str) -> Version:
+        """
+        Turns a string in the form of `5.X.Y | latest` into a semantic version.
+        If the string does not fit either of these forms, CliInputError will be raised.
+        If qt_ver == latest, and no versions exist corresponding to the filters specified,
+        then CliInputError will be raised.
+        If qt_ver == latest, and an HTTP error occurs, requests.RequestException will be raised.
+
+        :param qt_ver:  Either the literal string `latest`, or a semantic version
+                        with each part separated with dots.
+        """
+        assert qt_ver
+        if qt_ver == "latest":
+            latest_version = self.fetch_latest_version()
+            if not latest_version:
+                msg = "There is no latest version of Qt with the criteria '{}'".format(
+                    self.describe_filters()
+                )
+                raise CliInputError(msg)
+            return latest_version
+        version = helper.to_version(qt_ver)
+        if self.archive_id.is_major_ver_mismatch(version):
+            msg = "Major version mismatch between {} and {}".format(
+                self.archive_id.category, version
+            )
+            raise CliInputError(msg)
+        return version
+
+    @staticmethod
+    def fetch_http(rest_of_url: str) -> str:
+        base_urls = Settings.baseurl, random.choice(Settings.fallbacks)
+        for i, base_url in enumerate(base_urls):
+            try:
+                url = posixpath.join(base_url, rest_of_url)
+                return getUrl(
+                    url=url,
+                    timeout=(Settings.connection_timeout, Settings.response_timeout),
+                    logger=getLogger("aqt"),
+                )
```

Comment:
I've changed logger handling in master branch.
We don't pass logger as argument, but get it on each class.

Reply:
This part is missing from the refactoring, sorry.

Please see. https://github.com/miurahr/aqtinstall/pull/294

Reply:
Ok, this should be fixed now.